### ISSUE TITLE
false positive `reportUnsafeMultipleInheritance` when `__init__` call's `super`

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#623]
```py
class A:
    def __init__(self, *, a: int):
        super().__init__()


class B:
    def __init__(self):
        pass


class Something(A, B):  # error: Multiple inheritance is not allowed because the following base classes contain `__init__` or `__new__` methods that may not get called: B
    ...
```
[playground](https://basedpyright.com/?typeCheckingMode=all&code=MYGwhgzhAECCBcBYAUNN0AmBTAZtA%2BvgJYB2RALoQBQRYg4A00AVE2PNKeQJRKroCIAVwAOWAE5VuAOkKkK1bimXJQkGACE%2BA7HjllK%2BGnRy8UAgSPUqUaqNADKAewC2WcgAtSAcyqwmGrxoAMTQEuJO4hwAskIg5EQiIFicJB4SFGAkwClEMCRO5NBgICBOAO5YGNAARljAYEK00J4pOE6lFT61kCl2tDDATiTkYKTQAAb6CvgT0JGThCRY5YRzbp5OGDCeYEUuYACe0AVF3u7QDaVVHBrm6NKPKEA)

**Comment by DetachHead** [comment]
i originally didn't want to rely on checking the constructor because its implementation won't always be available eg. in stub files. but i guess there's no reason not to check it if it is available

**Comment by DetachHead** [comment]
it would also need to ensure that the `super().__init__` call matches the signature of the other class's constructor
